unit-tests/shared/src/test/require-jdk9-scala3/org/scalanative/testsuite/javalib/util/stream/BuildTestOnJDK9.scala

```diff
@@ -0,0 +1,23 @@
+package org.scalanative.testsuite.javalib.util.stream
+
+import java.util.stream._
+
+import java.{util => ju}
+import java.util.ArrayList
+import java.util.Map
+
+import java.util.stream.Collector.Characteristics
+
+import org.junit.Test
+import org.junit.Assert._
+
+class BuildTestOnJDK9 {
+
+  // Since: Java 9
+  @Test def executesOnJava9(): Unit = {
```

Comment:
It would be great to use any JDK version-specific API to make sure it actually is used and it compiles.
Also, probably 1 BuildTest would be enough, especially since we cannot be sure that they're actually included (we don't use verbose logging in tests so we it would not be logged either)

Reply:
I have logging turned on in my private build environment and visually 
determined that the three tests ran where they were supposed to run
and did not run where they were not, JDK 8.

I went with three to convince myself that tests beyond Java 9 were 
picked up (they are).  I only trust the numbers zero, one, two, and Infinity.
(Fi! on negative numbers, and do not get me started on NaN and negative 0.0)

I modeled the environment I expect to use in my pending CollectorsTestOnJDK* PR.
The BuildTests will have done their job by the time that PR is entered, so I envision
them being removed at that time.

I can remove two now if that simplifies this PR.